nmigen_boards/mercury.py

```diff
@@ -0,0 +1,226 @@
+            Attrs(IOSTANDARD="LVTTL", SLEW="FAST")
+        )
+    ]
+
+    # The "serial port" is in fact over SPI. The creators of the board provide
+    # a VHDL file for talking over this interface. In light of space
+    # constraints and the fact that both the FT245RL and FPGA can BOTH be
+    # SPI primaries, however, it may be necessary to sacrifice two "high-speed"
+    # (DIO, INPUT) pins instead.
+    serial = [
+        Resource("uart", 0,
+            Subsignal("tx", Pins("1", dir="o", conn=("dio", 0))),  # FTDI D1
+            Subsignal("rx", Pins("1", dir="i", conn=("input", 1))), # FTDI D0
+            Attrs(IOSTANDARD="LVCMOS33")
+        )
+    ]
+
+    # The remaining peripherals only make sense w/ the Baseboard installed.
+    # See: http://www.micro-nova.com/mercury-baseboard/
+    user_sw = [
+        Resource("user_sw", 0, Pins("1", dir="i", conn=("gpio", 0)),
+                 Attrs(IOSTANDARD="LVTTL")),
+        Resource("user_sw", 1, Pins("2", dir="i", conn=("gpio", 0)),
+                 Attrs(IOSTANDARD="LVTTL")),
+        Resource("user_sw", 2, Pins("3", dir="i", conn=("gpio", 0)),
+                 Attrs(IOSTANDARD="LVTTL")),
+        Resource("user_sw", 3, Pins("4", dir="i", conn=("gpio", 0)),
+                 Attrs(IOSTANDARD="LVTTL")),
+        Resource("user_sw", 4, Pins("5", dir="i", conn=("gpio", 0)),
+                 Attrs(IOSTANDARD="LVTTL")),
+        Resource("user_sw", 5, Pins("6", dir="i", conn=("gpio", 0)),
+                 Attrs(IOSTANDARD="LVTTL")),
+        Resource("user_sw", 6, Pins("7", dir="i", conn=("gpio", 0)),
+                 Attrs(IOSTANDARD="LVTTL")),
+        Resource("user_sw", 7, Pins("8", dir="i", conn=("gpio", 0)),
+                 Attrs(IOSTANDARD="LVTTL"))
+    ]
+
+    user_btn = [
+        Resource("user_btn", 1, Pins("1", dir="i", conn=("input", 0)),
+                 Attrs(IOSTANDARD="LVTTL")),
+        Resource("user_btn", 2, Pins("2", dir="i", conn=("input", 0)),
+                 Attrs(IOSTANDARD="LVTTL")),
+        Resource("user_btn", 3, Pins("3", dir="i", conn=("input", 0)),
+                 Attrs(IOSTANDARD="LVTTL")),
+        Resource("user_btn", 4, Pins("4", dir="i", conn=("input", 0)),
+                 Attrs(IOSTANDARD="LVTTL"))
+    ]
+
+    vga = [
+        Resource("vga_out", 0,
+            Subsignal("hsync", PinsN("led_0:3", dir="o")),
+            Subsignal("vsync", PinsN("led_0:4", dir="o")),
+
+            Subsignal("r", Pins("dio_0:1 dio_0:2 dio_0:3", dir="o")),
+            Subsignal("g", Pins("dio_0:4 dio_0:5 dio_0:6", dir="o")),
+            Subsignal("b", Pins("dio_0:7 clkio_0:1", dir="o")),
+            Attrs(IOSTANDARD="LVCMOS33", SLEW="FAST")
+        )
+    ]
+
+    extclk = [
+        Resource("extclk", 0, Pins("1", dir="i", conn=("clkio", 1)),
+                 Attrs(IOSTANDARD="LVCMOS33"))
+    ]
+
+    sevenseg = [
+        Resource("sevenseg", 0,
+                 Subsignal("a", PinsN("13", dir="o", conn=("gpio", 0))),
```

Comment:
This and the two following resources have wrong indentation.

Reply:
I accidentally forgot to remove the `check=True` argument, so I fixed and forced-pushed. Just FYI.